### PR TITLE
build: use Clang 15 for macOS cross-compilation

### PR DESCRIPTION
**Pull request by fanquake**
This will end up being a blocker for #28210, and is already part of #21778, even though an even newer LLVM/Clang combination is required (and still missing from upstream Guix). Seems straight-forward enough to just bump the macOS compiler to a more modern Clang.

### CODE REVIEW

**Comment by fanquake** [comment]
Rebased for #28328. I would slightly prefer this go-in before #27897.

**Comment by TheCharlatan** [comment]
Guix builds:

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
d7830926d348f3777cdd2e74118d3a51c96e864b47307ee45473423936c85f57  guix-build-832111ec916a/output/aarch64-linux-gnu/SHA256SUMS.part
65a2250e0abd1995d3726207b11a465b13b75d1883abf5ceaf1667338756114f  guix-build-832111ec916a/output/aarch64-linux-gnu/bitcoin-832111ec916a-aarch64-linux-gnu-debug.tar.gz
98f4f839ea331b4a15e32152bdf1da903ad1d1f8aa5752b2020269cdad6e5a9d  guix-build-832111ec916a/output/aarch64-linux-gnu/bitcoin-832111ec916a-aarch64-linux-gnu.tar.gz
fd594e12aae959078ab84e944f41ea16b8506a29b3f7067afd33e1489fc4c3bc  guix-build-832111ec916a/output/arm-linux-gnueabihf/SHA256SUMS.part
aa8d329225768c13b408efb2faefa87a5ff7db061b181c0fca7cb338f3d3f036  guix-build-832111ec916a/output/arm-linux-gnueabihf/bitcoin-832111ec916a-arm-linux-gnueabihf-debug.tar.gz
448a3fe6b01bbb391b44b8dc8c5e1bccc2a4d5a5132e427b84150cecb97f6413  guix-build-832111ec916a/output/arm-linux-gnueabihf/bitcoin-832111ec916a-arm-linux-gnueabihf.tar.gz
9ef6922d2d437f4092be564a081497e4b8d0fa3af8aa04065b64c27db2fd14b2  guix-build-832111ec916a/output/arm64-apple-darwin/SHA256SUMS.part
b60efb0671c07ee74db16baa6e46688837028dff5454ed8933ce623dea63a27b  guix-build-832111ec916a/output/arm64-apple-darwin/bitcoin-832111ec916a-arm64-apple-darwin-unsigned.dmg
77b7ed93ed713e56c49c81ae7da68e59c24e2238e9e5fbbc0c40f9ff5871ff8a  guix-build-832111ec916a/output/arm64-apple-darwin/bitcoin-832111ec916a-arm64-apple-darwin-unsigned.tar.gz
46ebc3dce256500fa534349ad3a0605858abc673e419e4793452a01dddd3d19a  guix-build-832111ec916a/output/arm64-apple-darwin/bitcoin-832111ec916a-arm64-apple-darwin.tar.gz
10b394010d4cc56c669972a6fe7fbae92bcc18e088accc1e735962006d16d282  guix-build-832111ec916a/output/dist-archive/bitcoin-832111ec916a.tar.gz
172bda90f8823ba9f98b5ad9d1b16c0f4f6870dc768c7d3bff358621b5a5599d  guix-build-832111ec916a/output/powerpc64-linux-gnu/SHA256SUMS.part
e07f5b5f891eccc7c180ef45559189d34d00f4e02b70b48c5fcdfe920ad18e9d  guix-build-832111ec916a/output/powerpc64-linux-gnu/bitcoin-832111ec916a-powerpc64-linux-gnu-debug.tar.gz
d53913167b1a9f2d4c7d0fa76b42db18a4d92e5cade703d6ec5e74ddc4f34e8c  guix-build-832111ec916a/output/powerpc64-linux-gnu/bitcoin-832111ec916a-powerpc64-linux-gnu.tar.gz
77b94c577ed3453a069e7f3e8a0d7026a2cc31a87300ea376ada62c4b40f86a8  guix-build-832111ec916a/output/powerpc64le-linux-gnu/SHA256SUMS.part
97f5a4d30748d375bd6c7dc2178a4b805fd580cc3ffb63b9f430f6f2300ef21d  guix-build-832111ec916a/output/powerpc64le-linux-gnu/bitcoin-832111ec916a-powerpc64le-linux-gnu-debug.tar.gz
55f2f92510acd7c62a98bc0faa69312fca9bb55610233a372c69353da38cebc6  guix-build-832111ec916a/output/powerpc64le-linux-gnu/bitcoin-832111ec916a-powerpc64le-linux-gnu.tar.gz
d941a324a4641371ebecea8873821775be7e23bdaadc8680dad6ace3f6ca3a2f  guix-build-832111ec916a/output/riscv64-linux-gnu/SHA256SUMS.part
a743489b5ebb27becf40326ee35b193c3a76947e55c1fe0b16b35c69c5daa25f  guix-build-832111ec916a/output/riscv64-linux-gnu/bitcoin-832111ec916a-riscv64-linux-gnu-debug.tar.gz
f73aa4ad0190f5588a2b1fd61375187982c3e7ff7275c3da65bdff76e7b44e4a  guix-build-832111ec916a/output/riscv64-linux-gnu/bitcoin-832111ec916a-riscv64-linux-gnu.tar.gz
783cb8362e184b26f447e5ee3e1d77d0bd9649d46eb59bbb8483cf950f9bc30f  guix-build-832111ec916a/output/x86_64-apple-darwin/SHA256SUMS.part
6a295f39a98d64ec49e395e35bfde4091023bca329143c34c62622b8cdafa49f  guix-build-832111ec916a/output/x86_64-apple-darwin/bitcoin-832111ec916a-x86_64-apple-darwin-unsigned.dmg
ca0d6763a6a5833051249bb0ba57ba8bfc7f075453c137609b0556036aff561a  guix-build-832111ec916a/output/x86_64-apple-darwin/bitcoin-832111ec916a-x86_64-apple-darwin-unsigned.tar.gz
fa3e2d853e1f0140c20b0f6f35969c55c9a9ebad716d5b3ec99580464f22882e  guix-build-832111ec916a/output/x86_64-apple-darwin/bitcoin-832111ec916a-x86_64-apple-darwin.tar.gz
e42f3ae704bdead771ad2340a3507b9ab2fcff27b744aa3cc50a17c40ff20c5a  guix-build-832111ec916a/output/x86_64-linux-gnu/SHA256SUMS.part
422e9037c47ea658ff33ac3160f37a394bc47e80d92cb78efc548594f5b127e7  guix-build-832111ec916a/output/x86_64-linux-gnu/bitcoin-832111ec916a-x86_64-linux-gnu-debug.tar.gz
650fdb98cc5d23e652e9571bedbb3b0fb9fd1b2cfe7a00794e995c8042cfcca2  guix-build-832111ec916a/output/x86_64-linux-gnu/bitcoin-832111ec916a-x86_64-linux-gnu.tar.gz
f6e3a9a62bc6c1a9781c89c556077f2ef6ede18cae53682b79e03132bdc5ba9a  guix-build-832111ec916a/output/x86_64-w64-mingw32/SHA256SUMS.part
3ddd9faeb586bf2d14fe71515eba0e707e5f9fafc0bf0716619cacad4295143c  guix-build-832111ec916a/output/x86_64-w64-mingw32/bitcoin-832111ec916a-win64-debug.zip
223224f697d1de9f13f947f5222c2f119c0cd9a631b5be24bb323e8354f569f2  guix-build-832111ec916a/output/x86_64-w64-mingw32/bitcoin-832111ec916a-win64-setup-unsigned.exe
b58a8f0345ae696e68b39c095b00e24379b1ec7d4b96066669360c3caec7defb  guix-build-832111ec916a/output/x86_64-w64-mingw32/bitcoin-832111ec916a-win64-unsigned.tar.gz
b2f9af5439a9a7a800d8561ffd740739b59dac936a1f17d8aecc9cacf8576d25  guix-build-832111ec916a/output/x86_64-w64-mingw32/bitcoin-832111ec916a-win64.zip
```

**Review by sedited — APPROVED** [review]
ACK 832111ec916a9245eac9fca9ca983b9b3d79f7d9

Tested guix-built binaries on macos 11.1 Big Sur. Also tested depends compilation. DId not spot any new build warnings.

File: depends/packages/native_clang.mk [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package=native_clang
-$(package)_version=11.1.0
+$(package)_version=15.0.6
```
Comment (hebasto) [inline depends/packages/native_clang.mk:2]: fa0fcb35f3d6c886dfc7af0e036fb1942650fca0:

> There is no x86_64 binaries for 15.0.7.

https://github.com/llvm/llvm-project/releases/download/llvmorg-15.0.7/clang+llvm-15.0.7-x86_64-apple-darwin21.0.tar.xz ?

Comment (fanquake) [reply depends/packages/native_clang.mk:2]: Those binaries are built for Darwin.

Comment (hebasto) [reply depends/packages/native_clang.mk:2]: Indeed. Sorry for noise.

File: depends/README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ The paths are automatically configured and no other options are needed unless ta
 
 #### For macOS cross compilation
 
-    sudo apt-get install curl bsdmainutils cmake libz-dev python3-setuptools libtinfo5 xorriso
+    sudo apt-get install curl bsdmainutils cmake libz-dev python3-setuptools libtinfo6 xorriso
```
Comment (hebasto) [inline depends/README.md:51]:
```
$ git grep -l libtinfo5
ci/test/00_setup_env_mac.sh
```

**Comment by hebasto** [comment]
Guix builds:
```
07edc86ee7b74f17744afbb1301cd080aa849220dcab068402c041b5fd524128  guix-build-ff42d81383d5/output/aarch64-linux-gnu/SHA256SUMS.part
c69a83c8a348bdeeb42cc9b543c0c1ae776efb104fa266a6e91fc82c02956997  guix-build-ff42d81383d5/output/aarch64-linux-gnu/bitcoin-ff42d81383d5-aarch64-linux-gnu-debug.tar.gz
9f26aa168238a22c15ffd14bae9bccc495a86777556d908a2bda58b076c1a2ba  guix-build-ff42d81383d5/output/aarch64-linux-gnu/bitcoin-ff42d81383d5-aarch64-linux-gnu.tar.gz
45181e6ab250245e3a51af3d6a978dece256b99a58c9bc227c9f89b619c1717b  guix-build-ff42d81383d5/output/arm-linux-gnueabihf/SHA256SUMS.part
11ea5a4154553745e0fcbdb7bdd8acf7ab47ac51de82632cf90f5d6e6a41c8ad  guix-build-ff42d81383d5/output/arm-linux-gnueabihf/bitcoin-ff42d81383d5-arm-linux-gnueabihf-debug.tar.gz
2348da028b271d98f9841de390dea177ac7bfd536d17c3ab0baa442e13324984  guix-build-ff42d81383d5/output/arm-linux-gnueabihf/bitcoin-ff42d81383d5-arm-linux-gnueabihf.tar.gz
f4a102b0c5886d59626122fe605514d9506c9fff8ae1274287c63712fbc41ac3  guix-build-ff42d81383d5/output/arm64-apple-darwin/SHA256SUMS.part
1e604950daece5aeb817b87535aff12cf68023fd416594d3b6ddad72af3d5f08  guix-build-ff42d81383d5/output/arm64-apple-darwin/bitcoin-ff42d81383d5-arm64-apple-darwin-unsigned.dmg
578c0dca66e438eed6ff37c390a9e4c2880a435e36fba1df1bb204978fd2956b  guix-build-ff42d81383d5/output/arm64-apple-darwin/bitcoin-ff42d81383d5-arm64-apple-darwin-unsigned.tar.gz
de18f253cf261eb80bc5b25915248e205251b8fb50d0e65aceb165f2973234e2  guix-build-ff42d81383d5/output/arm64-apple-darwin/bitcoin-ff42d81383d5-arm64-apple-darwin.tar.gz
5f4249d90a8e8be5d6dadd987f9d1304a03fff94630e9fbd197946a688039847  guix-build-ff42d81383d5/output/dist-archive/bitcoin-ff42d81383d5.tar.gz
6492d8233980cb512f24dfcfc52e8deff5df5a38c7ff07b01f8fb53eebee4054  guix-build-ff42d81383d5/output/powerpc64-linux-gnu/SHA256SUMS.part
366c7f2fe0802387f23a9b40cbaa98b3ba0b81f2ec757692f512daf654583d7e  guix-build-ff42d81383d5/output/powerpc64-linux-gnu/bitcoin-ff42d81383d5-powerpc64-linux-gnu-debug.tar.gz
dd33bff6ac6ed2dade6cf5034fb71fa2eb2dc9fe5af7ab2fad4e36ef6ed979b3  guix-build-ff42d81383d5/output/powerpc64-linux-gnu/bitcoin-ff42d81383d5-powerpc64-linux-gnu.tar.gz
4a9f71fc373879585da7695a4c5f37b610aaf19265d6e96b86371c4fd1fd6302  guix-build-ff42d81383d5/output/powerpc64le-linux-gnu/SHA256SUMS.part
cb621d18e1cc98bd953186db381660fb91ef52ca4026578fd4da23282d3d89fa  guix-build-ff42d81383d5/output/powerpc64le-linux-gnu/bitcoin-ff42d81383d5-powerpc64le-linux-gnu-debug.tar.gz
f511c27d82216acdc3b721dc012eb76216146ca09e9efbe1226506ee7b8f9a7e  guix-build-ff42d81383d5/output/powerpc64le-linux-gnu/bitcoin-ff42d81383d5-powerpc64le-linux-gnu.tar.gz
f2605cfc357bd0b37c638f3918c6c4784a48cc8538428fe4b6d8248f6827dc13  guix-build-ff42d81383d5/output/riscv64-linux-gnu/SHA256SUMS.part
2a66f5927fbfde8fcb1d0938132de7170f6c773497610753dd530ac382bec6a2  guix-build-ff42d81383d5/output/riscv64-linux-gnu/bitcoin-ff42d81383d5-riscv64-linux-gnu-debug.tar.gz
b318952ea3db16b343cc27c35d101d59902bcdcefc79ad0099722859788c6d2a  guix-build-ff42d81383d5/output/riscv64-linux-gnu/bitcoin-ff42d81383d5-riscv64-linux-gnu.tar.gz
90113fd9a5f54ce741057f4a17abea513eb16db0192732d7f0ef51a279b8cbd0  guix-build-ff42d81383d5/output/x86_64-apple-darwin/SHA256SUMS.part
ea203334502b229cc70945a9be54915a10dc135a89111cef0461f4e54d8afccf  guix-build-ff42d81383d5/output/x86_64-apple-darwin/bitcoin-ff42d81383d5-x86_64-apple-darwin-unsigned.dmg
900d904aca13a94f292c5c7b5ab0974253971af34e64fdd3edd44d5f74afb44b  guix-build-ff42d81383d5/output/x86_64-apple-darwin/bitcoin-ff42d81383d5-x86_64-apple-darwin-unsigned.tar.gz
7c2c8db41fbc7931aed8a9f09c9209bbd045089a5857986e48566910b896e2ec  guix-build-ff42d81383d5/output/x86_64-apple-darwin/bitcoin-ff42d81383d5-x86_64-apple-darwin.tar.gz
fa8acd4f62da754bd2302fa35b452522c41f2ff5948c78bf43f7a641ad79ff64  guix-build-ff42d81383d5/output/x86_64-linux-gnu/SHA256SUMS.part
87c4c22ab164a8ab654ce67f6cd9dde00959827e07a042f95cc02d2249b078a6  guix-build-ff42d81383d5/output/x86_64-linux-gnu/bitcoin-ff42d81383d5-x86_64-linux-gnu-debug.tar.gz
59493e490419cb50fbbad2a719d70da06fa062451667af947b58e9ae4746a9bb  guix-build-ff42d81383d5/output/x86_64-linux-gnu/bitcoin-ff42d81383d5-x86_64-linux-gnu.tar.gz
3faa8faed76662a682fcc2fb80a40027396101a398058ac3229f46e4018f7181  guix-build-ff42d81383d5/output/x86_64-w64-mingw32/SHA256SUMS.part
e206989b3ca6b91a01f4d6040fed9a5efb4736d503202dcd622e728c55d69dc5  guix-build-ff42d81383d5/output/x86_64-w64-mingw32/bitcoin-ff42d81383d5-win64-debug.zip
d7fe8deff67220ebbd9c90f539bede4530ce2412eeab11e149de2eb6991bc874  guix-build-ff42d81383d5/output/x86_64-w64-mingw32/bitcoin-ff42d81383d5-win64-setup-unsigned.exe
e3f429eea219ad4022974a1b737f0bcd123ab253b4c60ba0a89b97a8c1064e97  guix-build-ff42d81383d5/output/x86_64-w64-mingw32/bitcoin-ff42d81383d5-win64-unsigned.tar.gz
2466c8a3f5981bcda310220d3c49db59c17585215bd7a06b55c8a2a18b694e03  guix-build-ff42d81383d5/output/x86_64-w64-mingw32/bitcoin-ff42d81383d5-win64.zip
```

**Review by sedited — APPROVED** [review]
re-ACK ff42d81383d5555134ef2d760576b681cadc78d7

**Pull request merged**